Minimal config in installation guide

lachlanmacphee opened the pull request:
Hi all,

Firstly, thanks for such an amazing piece of software! I was going through the installation guide tonight to get my first Frigate setup going and encountered errors starting the container due to a missing config file, though this wasn't immediately clear, as I initially thought it was a `go2rtc` problem due to only seeing the bottom part of the logs in Portainer.

The error for the missing config wasn't **explicit** about it being a requirement too, it simply said it couldn't be found, so my assumption was a minimal config would be created automatically.

I think it'd be great to mention the requirement for a minimal configuration in order for the container to start. This is in the Getting Started guide already, but that's further down the page than Installation, so people might miss it.

<img width="292" alt="image" src="https://github.com/blakeblackshear/frigate/assets/104114845/32a0bdc4-b362-45b8-aa22-accb58e7dffc">

Specific changes:
+ Adds a subsection to the installation guide that mentions the requirement for a minimal config to start Frigate successfully.
+ Adds a warning not to proceed any further into the guide without this valid config

Potential concerns:
1. Two sources of truth for the minimal config - would it be better to link to the getting started guide?
2. Is the positioning of the warning and the subsection ideal?
3. Is the wording ideal? Does it create any potential confusion?

 I hope this PR helps. I did a search to see if there were any duplicates in PRs/Issues and couldn't find anything.

I apologise if I've missed something obvious and any of my assumptions here are incorrect, I come from a constructive place and just trying to make future Frigate installers' lives easier.

Regards,
Lachlan

NickM-27 commented:
This could maybe be pointed towards master but actually the next version of frigate (0.14) will create a minimal config file automatically if it does not exist

NickM-27 commented:
Also in general I think a link to the getting started guide would be better as that goes step by step through the minimal config creation. We wouldn't want _another_ config example to maintain

lachlanmacphee commented:
Thanks @NickM-27 , so just to clarify,

1. Would you like me to adjust the PR to point towards master?
2. Is this required if the next version will create it by default, not sure the release timeframe?

NickM-27 commented:
I think it may be better to remove the yaml example and instead say that a config is required, link to the getting started guide creating the config section. 

With this PR pointed towards master, we can later adjust it to say recommended instead of required before 0.14 releases.

lachlanmacphee committed:
Minimal config in installation guide (#11339)

+ Adds a subsection to the installation guide that mentions the requirement for a minimal config to start Frigate successfully.

+ Adds a warning not to proceed any further into the guide without this valid config

NickM-27 commented on docs/docs/frigate/installation.md:
```diff
@@ -47,12 +47,15 @@ services:
 
 :::
 
+### Creating a minimal configuration
+For Frigate to start, you will need to have a valid configuration file inside your `config/` directory mentioned in the storage section of this page. You can find a minimal reference configuration with a dummy camera on the [full reference config page](https://docs.frigate.video/configuration/reference/).
```
```suggestion
### Creating a minimal configuration

For Frigate to start, you will need to have a valid configuration file inside your `config/` directory mentioned in the storage section of this page. You can find a minimal reference configuration with a dummy camera on the [full reference config page](https://docs.frigate.video/configuration/reference/).
```

NickM-27 commented:
Will need to rebase 👍

lachlanmacphee commented:
Yep, on it now, apologies :)

lachlanmacphee commented:
Done, TIL the `--onto` flag!

NickM-27 commented on docs/docs/frigate/installation.md:
```diff
@@ -78,6 +81,12 @@ $ python -c 'print("{:.2f}MB".format(((1280 * 720 * 1.5 * 9 + 270480) / 1048576)
 
 The shm size cannot be set per container for Home Assistant add-ons. However, this is probably not required since by default Home Assistant Supervisor allocates `/dev/shm` with half the size of your total memory. If your machine has 8GB of memory, chances are that Frigate will have access to up to 4GB without any additional configuration.
 
+:::warning
```
This seems redundant to me given the first addition

lachlanmacphee replied:
My thinking was it's easy to gloss over those sections and doesn't hurt to have a warning banner? If disagree, happy to remove 👍

NickM-27 replied:
Right. Let's make the first one a warning instead of a header section and remove the second.

lachlanmacphee replied:
Sounds good 👍

lachlanmacphee committed:
Add spacing between Markdown title and text

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

lachlanmacphee committed:
Remove subsection in favour of exclusively adding a warning

NickM-27 commented on docs/docs/frigate/installation.md:
```diff
@@ -28,6 +28,12 @@ Frigate uses the following locations for read/write operations in the container.
 
+:::warning
+
+For Frigate to start, you will need to have a valid configuration file inside your `config/` directory mentioned above. You can find a minimal reference configuration with a dummy camera on the [full reference config page](https://docs.frigate.video/configuration/reference/).
```
```suggestion
For Frigate to start, it requires a valid configuration file inside the `/config/` directory mentioned above. There is a step by step guide to creating a minimal configuration in the [getting started guide](https://docs.frigate.video/guides/getting_started#configuring-frigate).
```

lachlanmacphee committed:
Adjust wording for the warning

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>
